haystack/nodes/reader/table.py

```diff
@@ -254,6 +371,27 @@ def _calculate_answer_offsets(answer_coordinates: List[Tuple[int, int]], table:
     def predict_batch(self, query_doc_list: List[dict], top_k: Optional[int] = None, batch_size: Optional[int] = None):
         raise NotImplementedError("Batch prediction not yet available in TableReader.")
 
+    class TapasForScoredQA(TapasPreTrainedModel):
+
+        def __init__(self, config):
+            super().__init__(config)
+
+            # base model
+            self.tapas = TapasModel(config)
+
+            # dropout (only used when training)
+            self.dropout = torch.nn.Dropout(config.hidden_dropout_prob)
+
+            # answer selection head
+            self.span_output_weights = torch.nn.Parameter(torch.zeros(2 * config.hidden_size))
+            self.span_output_bias = torch.nn.Parameter(torch.zeros([]))
+
+            # table scoring head
+            self.classifier = torch.nn.Linear(config.hidden_size, 2)
```

Comment:
Nice - looks very lean! If I get it right, we are still using the `from_pretrained()` model of the parent class (regular tapas) to load the weights for these extra layers? 
I guess the tricky part in the conversion was to get the naming of layers right (during export and now import) to make sure we load the right weights here?

Reply:
Yes, exactly.